src/components/CreditCardBilling/CreditCardBilling.tsx

```diff
@@ -0,0 +1,299 @@
+import React from 'react';
+import { scaleCanvas, setCanvasSize } from '@utils/canvas/canvasDimensions';
+import { clearCanvas } from '@utils/canvas/clearCanvas';
+import { css } from '@emotion/react';
+import { pallete } from './colorPallete';
+import { mockData } from './mockData';
+import { drawPopover } from './popOver';
+import type { Sides } from './types';
+
+export type Category = 'shopping' | 'transfer' | 'food';
+
+export interface CreditCardBilling {
+  category: Category;
+  value: number;
+}
+
+export interface CreditCardBillingWithCoords extends CreditCardBilling {
+  x: { start: number; end: number };
+  y: { start: number; end: number };
+  percentage: number;
+}
+
+interface CreditCardBillingProps {
+  width: React.CSSProperties['width'];
+  height: React.CSSProperties['height'];
+  data?: Record<'data', CreditCardBilling[]>;
+  margin?: Sides;
+}
+
+const Container = css`
+  position: relative;
+  outline: none;
+`;
+
+const Canvas = css`
+  position: absolute;
+  pointerevents: none;
+`;
+
+const roundRectRadii = 6;
+const defaultMargins = {
+  top: 40,
+  bottom: 40,
+  left: 20,
+  right: 20,
+};
+
+export default function CreditCardBilling({
+  data = mockData,
+  width,
+  height,
+  margin = defaultMargins,
+}: CreditCardBillingProps) {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const chartCanvasRef = React.useRef<HTMLCanvasElement>(null);
+  const popoverCanvasRef = React.useRef<HTMLCanvasElement>(null);
+  const popoverRAFID = React.useRef<number>(0);
```

Comment:
```suggestion
  const popoverRAFID = React.useRef(0);
```
마이너) 이 친구는 추론이 가능하겠네유